packages/flutter/lib/src/widgets/container.dart

```diff
@@ -247,35 +299,26 @@ class Container extends StatelessWidget {
   /// it would potentially result in the decoration drawing over the background
   /// color. To supply a decoration with a color, use `decoration:
   /// BoxDecoration(color: color)`.
-  Container({
+  const Container({
     super.key,
     this.alignment,
     this.padding,
     this.color,
     this.decoration,
     this.foregroundDecoration,
-    double? width,
-    double? height,
-    BoxConstraints? constraints,
+    this.width,
+    this.height,
+    this.constraints,
     this.margin,
     this.transform,
     this.transformAlignment,
     this.child,
     this.clipBehavior = Clip.none,
-  }) : assert(margin == null || margin.isNonNegative),
-       assert(padding == null || padding.isNonNegative),
-       assert(decoration == null || decoration.debugAssertIsValid()),
-       assert(constraints == null || constraints.debugAssertIsValid()),
-       assert(decoration != null || clipBehavior == Clip.none),
```

Comment:
Moving these asserts will make it harder to debug code because incorrect values are not surfaced anymore right when the widget is instantiated (with a stack trace that likely points exactly to where you pass in the incorrect value). Instead, the exception is surfaced during build with a framework-internal stack trace that will make it harder to relate the exception back to your own code.

Reply:
I personally would rather have `const` Containers, even if these asserts are moved into the build method and thus aren't quite as easy to understand. (Hopefully it's not quite as difficult now that they're in the Container build method instead of a helper class.)

Reply:
Yeah, I still think that this isn't great. It will make debugging pretty painful. And if you absolutely need `const`s there is a work around: Instead of using a container use its building blocks (Padding, Decoration, etc) directly. You'll probably get a better performance boost out of that because you can skip all the logic in container where it is figuring out which building blocks to use.

Reply:
Additionally, this also seems to be a breaking change looking at the failing customer tests. I don't think this change is worth the trouble.